diff --git a/anvil.yaml b/anvil.yaml
index 329081e..0cbc547 100644
--- a/anvil.yaml
+++ b/anvil.yaml
@@ -45,6 +45,7 @@ runtime_options:
   legacy_features: {__dict__: true, bootstrap3: true, class_names: true, root_container: true}
   server_spec: {base: python310-datascience}
   server_version: python3-sandbox
+  server_persist: true
   version: 3
 services:
 - client_config: {}
diff --git a/client_code/Base/__init__.py b/client_code/Base/__init__.py
index ec205ce..2dd6ca1 100644
--- a/client_code/Base/__init__.py
+++ b/client_code/Base/__init__.py
@@ -8,6 +8,7 @@ from anvil.tables import app_tables
 from anvil.js.window import jQuery, innerWidth
 from anvil.js import get_dom_node
 
+set_default_error_handling(print)
 
 class Base(BaseTemplate):
   def __init__(self, **properties):
diff --git a/server_code/Global_Server_Functions/__init__.py b/server_code/Global_Server_Functions/__init__.py
index e5c123a..2bd33a0 100644
--- a/server_code/Global_Server_Functions/__init__.py
+++ b/server_code/Global_Server_Functions/__init__.py
@@ -1,3 +1,5 @@
+import os
+from dotenv import load_dotenv
 import anvil.files
 from anvil.files import data_files
 import anvil.tables as tables
@@ -6,15 +8,18 @@ from anvil.tables import app_tables
 import anvil.server
 from collections.abc import Iterable
 import pandas as pd
+from sqlalchemy import create_engine
 
+load_dotenv()
 
+engine = create_engine(os.getenv("SQL_CONNECTION"))
 _DATA_CACHE = None
 
 def get_data(project_privacy=False):
   global _DATA_CACHE
   # Load data once
   if _DATA_CACHE is None:
-    _DATA_CACHE = pd.read_pickle(data_files['synthetic_data.pkl'])
+    _DATA_CACHE = pd.read_sql_table("app_data", con=engine)
 
   df = _DATA_CACHE  # start from cached data
 
